qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -414,10 +424,54 @@ def open_file(self):
             return
         self.editor.open_file_in_new_tab(filepath)
 
-    def save_file(self):
-        # todo: how should this interact with project saving and workspaces when they are implemented?
+    def save_script(self):
         self.editor.save_current_file()
 
+    def save_project(self):
+        if self.last_project_location is None:
+            self.save_project_as()
+        else:
+            # Clear directory before saving to remove old workspaces
+            files = glob.glob(self.last_project_location + '/.*')
```

Comment:
I feel like we should be a little more restrictive over what we delete. Do we currently keep track of the workspaces /fileswe saved in a  project? How about we keep track of the files we saved in the project, this would be ideal to be kept in the new `Project` class, and only remove those that we no longer want. I think this is safer than glob removing everything.

Reply:
With the files that we have saved in a project it's quite a simple call to that projects location with the os module's listdir function, currently we delete everything because well we are replacing everything and everything in theory should be deleted. In no one should ever be able to delete anything that isn't inside this folder that they are using for saving into either.

Reply:
If someone saved their own script into that directory we would then remove that too. I really don't think we should be removing things that we did not save ourselves unless asked by a user. It's not about how simple the code is to write but the potential to lose data inadvertently.

Reply:
I've wrote a new function for handling clearing of workspaces that are not going to be overwritten, but not touching an scripts. You made a very valid point :+1: